src/snowflake/snowpark/session.py

```diff
@@ -1223,7 +1242,7 @@ def _get_dependency_packages(
         elif len(errors) > 0:
             raise RuntimeError(errors)
 
-        dependency_packages: List[pkg_resources.Requirement] = []
+        dependency_packages: Optional[List[pkg_resources.Requirement]] = []
```

Comment:
I don't love that this is allowed to be `None` here, but at the end you assert that it isn't. Could we please introduce an intermittent variable for this?